langchain/scripts/identify-secrets.js

```diff
@@ -0,0 +1,77 @@
+import ts from "typescript";
+import * as fs from "fs";
+
+export function identifySecrets() {
+  const secrets = new Set();
+
+  const tsConfig = ts.parseJsonConfigFileContent(
+    ts.readJsonConfigFile("./tsconfig.json", (p) =>
+      fs.readFileSync(p, "utf-8")
+    ),
+    ts.sys,
+    "./src/"
+  );
+
+  for (const fileName of tsConfig.fileNames.filter(
+    (fn) => !fn.endsWith("test.ts")
+  )) {
+    const sourceFile = ts.createSourceFile(
```

Comment:
Dang this is gnarly - purpose is to look for required env variables in source files?

Reply:
To look for the secret ids defined in Serializable classes yes

Reply:
Might be good to refactor into different methods for clarity